pkg/component/etcd/etcd/crd.go

```diff
@@ -9,98 +9,146 @@ import (
-		r := resource
+	for _, resource := range c.crdGetter.GetAllCRDs() {
+		r := resource.DeepCopy()
 		fns = append(fns, func(ctx context.Context) error {
-			return c.applier.ApplyManifest(ctx, kubernetes.NewManifestReader([]byte(r)), kubernetes.DefaultMergeFuncs)
+			_, err := controllerutil.CreateOrPatch(ctx, c.client, r, nil)
+			return err
 		})
 	}
-
 	return flow.Parallel(fns...)(ctx)
 }
 
 func (c *crd) Destroy(ctx context.Context) error {
-	etcdList := &druidv1alpha1.EtcdList{}
+	etcdList := &druidcorev1alpha1.EtcdList{}
 	// Need to check for both error types. The DynamicRestMapper can hold a stale cache returning a path to a non-existing api-resource leading to a NotFound error.
 	if err := c.client.List(ctx, etcdList); err != nil && !meta.IsNoMatchError(err) && !apierrors.IsNotFound(err) {
 		return err
 	}
 
 	if len(etcdList.Items) > 0 {
-		return fmt.Errorf("cannot delete etcd CRDs because there are still druidv1alpha1.Etcd resources left in the cluster")
+		return fmt.Errorf("cannot delete etcd CRDs because there are still druidcorev1alpha1.Etcd resources left in the cluster")
 	}
 
-	if err := gardenerutils.ConfirmDeletion(ctx, c.client, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: etcdCRDName}}); client.IgnoreNotFound(err) != nil {
+	if err := gardenerutils.ConfirmDeletion(ctx, c.client, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: druidcorecrds.ResourceNameEtcd}}); client.IgnoreNotFound(err) != nil {
 		return err
 	}
 
-	etcdCopyBackupsTaskList := &druidv1alpha1.EtcdCopyBackupsTaskList{}
+	etcdCopyBackupsTaskList := &druidcorev1alpha1.EtcdCopyBackupsTaskList{}
 	if err := c.client.List(ctx, etcdCopyBackupsTaskList); err != nil && !meta.IsNoMatchError(err) && !apierrors.IsNotFound(err) {
 		return err
 	}
 
 	if len(etcdCopyBackupsTaskList.Items) > 0 {
-		return fmt.Errorf("cannot delete etcd CRDs because there are still druidv1alpha1.EtcdCopyBackupsTask resources left in the cluster")
+		return fmt.Errorf("cannot delete etcd CRDs because there are still druidcorev1alpha1.EtcdCopyBackupsTask resources left in the cluster")
 	}
 
-	if err := gardenerutils.ConfirmDeletion(ctx, c.client, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: etcdCopyBackupsTaskCRDName}}); client.IgnoreNotFound(err) != nil {
+	if err := gardenerutils.ConfirmDeletion(ctx, c.client, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: druidcorecrds.ResourceNameEtcdCopyBackupsTask}}); client.IgnoreNotFound(err) != nil {
 		return err
 	}
 
 	var fns []flow.TaskFn
 
-	for _, resource := range crdResources {
+	for _, resource := range c.crdGetter.GetAllCRDs() {
 		r := resource
 		fns = append(fns, func(ctx context.Context) error {
-			return client.IgnoreNotFound(c.applier.DeleteManifest(ctx, kubernetes.NewManifestReader([]byte(r))))
+			return client.IgnoreNotFound(c.client.Delete(ctx, r, &client.DeleteOptions{}))
```

Comment:
You don't need to pass an empty DeleteOption, it is varargs - it can stay empty. Or said with other words, it is optional.

```suggestion
			return client.IgnoreNotFound(c.client.Delete(ctx, r))
```